components/eamxx/src/share/tests/eamxx_time_interpolation_tests.cpp

```diff
@@ -0,0 +1,404 @@
+#include <catch2/catch.hpp>
+
+#include "share/grid/mesh_free_grids_manager.hpp"
+
+#include "share/field/field_utils.hpp"
+#include "share/field/field.hpp"
+#include "share/field/field_manager.hpp"
+
+#include "share/util/eamxx_time_interpolation.hpp"
+#include "share/util/scream_setup_random_test.hpp"
+#include "share/util/scream_time_stamp.hpp"
+
+#include "share/io/scream_output_manager.hpp"
+
+#include "ekat/ekat_parameter_list.hpp"
+/*-----------------------------------------------------------------------------------------------
+ * Test TimeInterpolation class
+ *-----------------------------------------------------------------------------------------------*/
+
+namespace scream {
+
+// Test Constants
+constexpr Real tol            = std::numeric_limits<Real>::epsilon()*1e5;
+constexpr int  slp_switch     = 4; // The frequency that we change the slope of the data written to file.
+constexpr int  dt             = 100;
+constexpr int  total_snaps    = 10;
+constexpr int  snap_freq      = 4;
+constexpr int  slope_freq     = snap_freq; // We will change the slope every slope_freq steps to ensure that the data is not all on the same line.
+constexpr int  snaps_per_file = 3;
+
+// Functions needed to set the test up
+std::shared_ptr<const GridsManager> get_gm (const ekat::Comm& comm, const int ncols, const int nlevs);
+std::shared_ptr<FieldManager> get_fm (const std::shared_ptr<const AbstractGrid>& grid, const util::TimeStamp& t0, const int seed);
+util::TimeStamp init_timestamp();
+std::vector<std::string> create_test_data_files(const ekat::Comm& comm, const std::shared_ptr<const GridsManager>& gm,const util::TimeStamp& t0, const int seed);
+
+// Functions needed to run the test
+void update_field_data(const Real slope, const Real dt, Field& field);
+bool views_are_approx_equal(const Field& f0, const Field& f1, const Real tol);
+
+// Helper randomizer
+auto my_pdf = [&](std::mt19937_64& engine) -> Real {
```

Comment:
Since you're at file scope, this can simply be a normal function, rather than a lambda...